design/034-new_lockfiles.md

```diff
@@ -0,0 +1,237 @@
+# Proposal: New Conan 2.0 lockfiles
+
+| **Status**        |                                                   |
+|:------------------|:--------------------------------------------------|
+| **RFC #**         | [034](https://github.com/conan-io/tribe/pull/34)  |
+| **Submitted**     | 2022-04-26                                        |
+| **Tribe votes**   |                                                   |
+
+
+## Summary
+
+Conan 2.0 will implement completely new lockfiles, with a new proposal of format, definition, behavior, usage and flow, based on the following principles:
+
+- Lockfiles files will be json files containing only three lists of ordered references. One list for “host” requires, another list for “build” requires, another list for “python” requires.
+- Lockfiles will no longer contain information about profiles, settings, options (as they did in 1.X)
```

Comment:
I know that profiles can be distributed with ``conan config install``, but it is not always obvious which profiles or overrides were used in the build.

E.g. now a CI job produces lockfile as an artifact which can be consumed later. Especially in "testing" or "deploy" jobs which call ``conan lockfile install``. This information will be unavailable, right?

Basically, what I'm looking for is a way to avoid discrepancy between which environment was used to build and to consume. Conan already have composable profiles and overrides via settings and options, and in our CI more overrides can come via environment variables or extra job parameters (e.g. if one wants an experimental build with specific settings).

I suspect is would be possible to implement this on top of Conan with custom CI scripts (e.g. create a profile file by aggregating data which otherwise would have been passed on the command line and pass only it as a single ``--profile`` argument and export as a CI job artifact), but maybe it's worth doing similar thing right in Conan. Like a "profile lock" which (if specified) would complement dependencies' lockfiles and have similar strict checks as with current lockfiles.

Reply:
I am happy to see I am not the only one who is concerned about preserving profile information
(what I did and do so far via having them in the lockfile, and that correct and for free)

Reply:
I would also second the need to be able to capture/obtain the full settings/options context in order to be able to reconstruct the environment from binary package creation time.

Reply:
I actually realized one more use case for that: sometimes we use same profile name which is specific to particular docker image. Similar to preparing image with a compiler and then running ``conan profile new default --detect``. Then we build using that image and profile.